editor/filesystem_dock.cpp

```diff
@@ -3861,8 +3861,20 @@ void FileSystemDock::load_layout_from_config(Ref<ConfigFile> p_layout, const Str
 
 	if (p_layout->has_section_key(p_section, "dock_filesystem_selected_paths")) {
 		PackedStringArray dock_filesystem_selected_paths = p_layout->get_value(p_section, "dock_filesystem_selected_paths");
+		PackedStringArray dock_filesystem_really_selected_paths;
```

Comment:
It would be more efficient to instead keep a list of deleted paths. It's unlikely that a path is going to be invalid.